src/csvcubedpmd/csvwcli/pull.py

```diff
@@ -134,20 +135,24 @@ def _get_rdf_file_dependencies(metadata_file: str) -> List[str]:
 
 
 def _get_table_group_for_metadata_file(metadata_file: str) -> Dict:
-    if isinstance(metadata_file, str):
+    if isinstance(metadata_file, str) and _looks_like_uri(metadata_file):
         return requests.get(metadata_file).json()
     else:
-        with open(metadata_file.absolute(), "r") as f:
+        with open((Path(metadata_file)).absolute(), "r") as f:
             return json.load(f)
 
 
 def pull(csvw_metadata_url: str, output_dir: Path) -> None:
     """
     Pull all of the relative dependencies of a CSV-W metadata file down to the :obj:`output_dir`.
     """
-    csvw_metadata_file_name = _get_file_name_from_url(csvw_metadata_url)
-    _ensure_dir_structure_exists(output_dir)
-    _download_to_file(csvw_metadata_url, output_dir / csvw_metadata_file_name)
+    if _looks_like_uri(csvw_metadata_url):
+        csvw_metadata_file_name = _get_file_name_from_url(csvw_metadata_url)
+        _ensure_dir_structure_exists(output_dir)
```

Comment:
we're doing this in both conditions, just pull this line out and put it once before the if.